mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,803 @@
+                    }
+                }
+
+                try_or_return_stopped_as_err!(
+                    storage
+                        .add_transactions_to_history(&coin.history_wallet_id(), tx_details)
+                        .await
+                        .map_err(|e| format!("{:?}", e)),
+                    StopReason::StorageError,
+                    "add_transactions_to_history failed"
+                );
+
+                iterate_more = (TX_PAGE_SIZE as u32 * page) < response.total_count;
+                page += 1;
+            }
+
+            Ok(highest_height)
+        }
+
+        let q = format!(
+            "coin_spent.spender = '{}' AND tx.height > {}",
+            self.address.clone(),
+            self.from_block_height
+        );
+        let highest_send_tx_height = match fetch_and_insert_txs(
+            &self.tendermint_assets_conf,
+            self.address.clone(),
+            &ctx.coin,
+            &ctx.storage,
+            q,
+            self.from_block_height,
+        )
+        .await
+        {
+            Ok(block) => block,
+            Err(stopped) => {
+                if let StopReason::RpcClient(e) = &stopped.stop_reason {
+                    log::error!("Sent tx history process turned into cooldown mode due to rpc error: {e}");
+                    return Self::change_state(OnIoErrorCooldown::new(
+                        self.address.clone(),
+                        self.from_block_height,
+                        self.tendermint_assets_conf,
+                    ));
+                }
+
+                return Self::change_state(stopped);
+            },
+        };
+
+        let q = format!(
+            "coin_received.receiver = '{}' AND tx.height > {}",
+            self.address.clone(),
+            self.from_block_height
+        );
+        let highest_recieved_tx_height = match fetch_and_insert_txs(
+            &self.tendermint_assets_conf,
+            self.address.clone(),
+            &ctx.coin,
+            &ctx.storage,
+            q,
+            self.from_block_height,
+        )
+        .await
+        {
+            Ok(block) => block,
+            Err(stopped) => {
+                if let StopReason::RpcClient(e) = &stopped.stop_reason {
+                    log::error!("Received tx history process turned into cooldown mode due to rpc error: {e}");
+                    return Self::change_state(OnIoErrorCooldown::new(
+                        self.address.clone(),
+                        self.from_block_height,
+                        self.tendermint_assets_conf,
+                    ));
+                }
+
+                return Self::change_state(stopped);
+            },
+        };
+
+        let last_fetched_block = cmp::max(highest_send_tx_height, highest_recieved_tx_height);
+
+        log::info!(
+            "Tx history fetching finished for tendermint. Last fetched block {}",
+            last_fetched_block
+        );
```

Comment:
With the current message, logs for different coins will be indistinguishable from each other. It's only possible to guess the coin by block number. Could you please add the platform ticker to the message?